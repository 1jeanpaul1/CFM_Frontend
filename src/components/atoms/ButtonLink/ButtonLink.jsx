import PropTypes from 'prop-types';
import { Button } from '@mui/material';
import { NextLink } from 'components/atoms';

export default function ButtonLink(props) {
  return <Button component={NextLink} {...props} />;
}

const toPathname = PropTypes.shape({
  pathname: PropTypes.string.isRequired,
  query: PropTypes.object.isRequired,
});

ButtonLink.propTypes = {
  /**
   * The URL or pathname object to navigate to.
   *
   * to='/about'
   * to={{ pathname: '/blog/[slug]', query: { slug: post.slug }, }}
   */
  to: PropTypes.oneOfType([PropTypes.string, toPathname]).isRequired,

  /**
   * Text describing the content of the link.
   * eg: aria-label='Learn more about the Fridge Finder project.'
   */
  'aria-label': PropTypes.string.isRequired,

  /***
   * Style of button from theme
   */
  variant: PropTypes.oneOf(['outlined', 'contained']).isRequired,
};
