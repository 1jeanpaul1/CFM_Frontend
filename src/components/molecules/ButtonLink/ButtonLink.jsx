import PropTypes from 'prop-types';
import { Button } from '@mui/material';
import { NextLinkAnchor } from 'components/atoms';

export default function ButtonLink(props) {
  return <Button component={NextLinkAnchor} {...props} />;
}

const toPathname = PropTypes.shape({
  pathname: PropTypes.string.isRequired,
  query: PropTypes.object.isRequired,
});

ButtonLink.propTypes = {
  /**
   * The URL or pathname object to navigate to.
   *
   * to='/about'
   * to={{ pathname: '/blog/[slug]', query: { slug: post.slug }, }}
   */
  to: PropTypes.oneOfType([PropTypes.string, toPathname]).isRequired,

  /**
   * Text describing the content of the link.
   * eg: aria-label='Learn more about the community fridge map project.'
   */
  ariaLabel: PropTypes.string.isRequired,

  /***
   * Style of button from theme
   */
  variant: PropTypes.oneOf(['outlined', 'contained']).isRequired,
};
