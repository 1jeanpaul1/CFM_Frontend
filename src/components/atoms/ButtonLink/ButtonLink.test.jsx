import { render } from '@testing-library/react';
import ButtonLink from './ButtonLink';

describe('ButtonLink', () => {
  const renderButtonLink = (props) => render(<ButtonLink {...props} />);

  it('should render the correct props', () => {
    const button = renderButtonLink({
      to: '/about',
      'aria-label': 'Learn more',
      variant: 'outlined',
    }).getByRole('link');
    expect(button).toHaveAttribute('href', '/about');
    expect(button).toHaveAttribute('aria-label', 'Learn more');
    expect(button).toHaveClass('MuiButton-outlined');
  });
});
