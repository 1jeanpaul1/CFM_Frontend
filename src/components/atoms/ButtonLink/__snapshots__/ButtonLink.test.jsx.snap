// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink should render the ButtonLink component correctly 1`] = `
<a
  aria-label="Learn more"
  class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
  href="/about"
  tabindex="0"
>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;
